corehq/apps/sso/tests/test_backends.py

```diff
@@ -254,6 +254,81 @@ def test_new_user_created_and_data_is_saved(self):
             ]
         )
 
+    def test_new_user_displayname_is_used_if_first_and_last_are_missing(self):
```

Comment:
These new tests have a lot of code just to test that the names are grabbed appropriately. One way to shrink the tests and expose just the relevant information would be to create helper methods for the irrelevant details (i.e. the contents of `cleaned_data` don't matter for these tests). 

However, I think a better approach might be to make these tests lower level. It's great that there is a happy-path test to ensure that all the components integrate with each other, but I don't think its necessary to need an `AsyncSignupRequest` when we're testing edge cases within `get_sso_user_first_name_from_session`. 

With integration tests, beyond being more expensive, a developer needs full context to understand what is going on. The location of where those tests are, and what they should cover is ambiguous. If these tests were instead replaced with unit tests, then those problems are solved:
- I can look exclusively at `get_sso_user_first_name_from_session()` to understand the test.
- because `get_sso_user_first_name_from_session()` lives in _session_helpers.py_, it is natural that its tests live in _test_session_helpers.py_
- There is no ambiguity on what should be tested. If there is logic within `get_sso_user_first_name_from_session()`, a test is expected to verify that logic.